terraform/setup/setup.tf

```diff
@@ -224,19 +224,22 @@ resource "aws_ecr_repository" "mocked_server_ecr_repo" {
 #Document: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket
 resource "aws_s3_bucket" "terrafrom-state" {
   bucket = module.common.terraform_state_s3_bucket_name
+}
 
-
-  #Allow multiple upload
-  versioning {
-    enabled = true
+resource "aws_s3_bucket_versioning" "terrafrom-state-versioning" {
+  bucket = aws_s3_bucket.terrafrom-state.id
+  versioning_configuration {
+    status = "Enabled"
   }
+}
```

Comment:
Is the purpose of this bucket to house the terraform state file? If so there is a better way to implement this using the terraform backend module. https://www.terraform.io/language/settings/backends

Reply:
I now see the commented code out on top. I am unsure why that's not being used.